### PR TITLE
OSP-191: display a nice to read error when neutron is not configured properly

### PR DESCRIPTION
**Pull request by wolverineav**
Reviewer: @sarath-kumar @WeifanFu-bsn 

 - some of our horizon plugin features depend on bsn_service_plugin
   being enabled in neutron server
 - add a check when creating objects to see if the plugin is enabled
 - if not, raise an exception that can be displayed on horizon gui

### CODE REVIEW

**Comment by bsn-bot3** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.bigswitch.com/job/openstack_horizon_bsn_pull_req/179/

**Comment by bsn-bot3** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://jenkins.bigswitch.com/job/openstack_horizon_bsn_pull_req/180/

File: horizon_bsn/api/neutron.py [hunk 1]
```diff
@@ -98,6 +132,10 @@ def reachabilitytest_create(request, **params):
     :param expected_result: expected result of the reachability test
     """
     LOG.debug("reachabilitytest_create(): params=%s", params)
```
Comment (WeifanFu-bsn) [inline horizon_bsn/api/neutron.py]: Do we need to change the log or skip it when it is not supported? Might not be important since it is debug level log.

Comment (wolverineav) [reply horizon_bsn/api/neutron.py]: i haven't found the logfile or command to see this log. usually all upstream APIs have a debug log before doing anything. so i kept it.
shouldn't be an issue though.

File: horizon_bsn/api/neutron.py [hunk 2]
```diff
@@ -24,6 +27,37 @@
 neutronclient = neutron.neutronclient
 
 
+@memoized
+def list_extensions(request):
```
Comment (WeifanFu-bsn) [inline horizon_bsn/api/neutron.py]: I found https://github.com/openstack/horizon/blob/master/openstack_dashboard/api/neutron.py#L1808

The codes looks identical, could we use it directly, or is there any reason we need to duplicate the codes here?

Comment (wolverineav) [reply horizon_bsn/api/neutron.py]: yep, its the same check. if upstream moves api.neutron to a new location, it causes failure at our end without notice.
for minor util methods, its easier to clone and not worry about future updates and breakages

Comment (wolverineav) [reply horizon_bsn/api/neutron.py]: correction - i did not realize i'm already importing api.neutron for creating the neutronclient.
i'm not saving anything by cloning the method from the same class 🤦🏻‍♂️
now uses upstream method directly 🙂

**Pull request merged**